components/Header.tsx

```diff
@@ -1,74 +1,130 @@
 // Copyright 2023 the Deno authors. All rights reserved. MIT license.
 import {
   ACTIVE_LINK_STYLES,
-  BUTTON_STYLES,
   LINK_STYLES,
-  NAV_STYLES,
   SITE_BAR_STYLES,
   SITE_NAME,
 } from "@/utils/constants.ts";
 import { stripe } from "@/utils/payments.ts";
-import { Bell, CircleFilled } from "./Icons.tsx";
-import { getToggledStyles } from "@/utils/display.ts";
+import { Bars, Bell, CircleFilled, Cross } from "./Icons.tsx";
+import { cx } from "@twind/core";
+import type { JSX } from "preact";
+
+interface MenuItemProps {
+  name: string | JSX.Element;
+  href: string;
+  active?: boolean;
+  class?: string;
+  ariaLabel?: string;
+}
+
+function MenuItem(props: MenuItemProps) {
+  return (
+    <a
+      href={props.href}
+      class={cx(
+        props.active ? ACTIVE_LINK_STYLES : LINK_STYLES,
+        "relative text-gray-500 px-3 py-4 sm:py-2",
+        props.class,
+      )}
+      aria-label={props.ariaLabel}
+    >
+      {props.name}
+    </a>
+  );
+}
```

Comment:
I think it'd be cleaner if we just wrote the menu HTML without iterating over an array of elements. Menu items can vary in requirements, making a flexible `<MenuItem />` challenging to manage past a certain point. We can have a `MENU_ITEM_STYLES` constant for any common styles between these menu items. WDYT?